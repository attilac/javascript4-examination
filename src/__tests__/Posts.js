import React from 'react';
import { mount } from 'enzyme';
import Posts from '../components/Posts';

function postMockPost(){
  const mockPost = JSON.stringify(
    [
      { 
        id: '1', 
        title: 'testTitle',
        content: 'testContent',
        author: 'testAuthor',
        date: '2018-1-23 14:44:38'
      },
    ]);
  localStorage.setItem('posts', mockPost);
};
  

beforeEach(() => {
  postMockPost();
});

afterEach(() => {
  localStorage.clear();
});

it('should fetch post as state on mount', ()=> {
  const wrapper = mount(<Posts currentPersona="testAuthor" /> );
  expect(wrapper.state().posts[0].title).toEqual('testTitle');
});

// EXEMPEL 1
// Enhetstest som automatiskt blir integrationstest. Testar 3 funktioner indirekt.
it('should remove post from state', ()=> {
  const wrapper = mount(<Posts currentPersona="testAuthor" /> );
  expect(wrapper.state().posts).toHaveLength(1); // test before
  wrapper.instance().removePost('1');
  expect(wrapper.state().posts).toHaveLength(0); // test after
});

it('should render my post & author', () => {
  const wrapper = mount(<Posts currentPersona="testAuthor" /> );
  expect(wrapper.find('article h2').text()).toEqual('testTitle');
  expect(wrapper.find('article div p').text()).toEqual('testContent');
});

it('should match snapshot', () => {
  const wrapper = mount(<Posts currentPersona="testAuthor" /> );
  expect(wrapper.html()).toMatchSnapshot();
});