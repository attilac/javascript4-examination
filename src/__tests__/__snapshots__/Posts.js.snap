// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `"<div class=\\"flex flex-wrap mx-auto pt-8\\"><form class=\\"container mx-auto flex flex-col p-6\\"><label for=\\"title\\" class=\\"mb-2\\">Title</label><input type=\\"text\\" value=\\"\\" name=\\"title\\" id=\\"title\\" class=\\"shadow rounded border mb-2 p-2\\"><label for=\\"content\\" class=\\"mb-2\\">Content </label><textarea id=\\"content\\" name=\\"content\\" class=\\"shadow rounded border mb-2 p-2\\"></textarea><input type=\\"submit\\" value=\\"Create\\" class=\\"bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full\\"></form><div class=\\"w-full shadow p-6 m-6 border rounded relative\\"><article class=\\"mb-2\\"><button class=\\"bg-red-dark hover:bg-red-darker text-white font-bold py-2 px-4 rounded-full absolute pin-t pin-r mx-4 my-6\\" data-test=\\"button\\">x</button><h2 class=\\"mb-4 w-4/5\\">testTitle</h2><div class=\\"mb-4 text-grey-darkest markdown\\"><p>testContent</p></div><p class=\\"text-grey-dark mb-4\\">Posted by: testAuthor @ 2018-1-23 14:44:38</p></article><div class=\\"py-2\\"><h2 class=\\"text-indigo-darker border-b mb-4\\">Comments</h2><form class=\\"container mx-auto flex flex-col p-6\\"><label for=\\"comment\\" class=\\"mb-2\\">Comment</label><textarea id=\\"comment\\" name=\\"comment\\" class=\\"shadow rounded border mb-2 p-2\\"></textarea><input type=\\"submit\\" value=\\"Comment\\" class=\\"bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full\\"></form></div></div></div>"`;
